### PR TITLE
(fix) integration with backend signup

### PR DESCRIPTION
**Pull request by sarah-kamall**

### CODE REVIEW

File: app/src/data/icons.tsx [hunk 1]
```diff
@@ -128,13 +128,13 @@ const iconMap: { [K in iconStrings]: React.ReactNode } = {
   ),
   Show: (
     <VisibilityIcon
-      fontSize="large"
```
Comment (Asmaa-204) [inline app/src/data/icons.tsx:131]: finally back to small eyes🥹

Comment (sarah-kamall) [reply app/src/data/icons.tsx:131]: m2drsh arfodlk tlb

File: app/src/features/authentication/login/hooks/useAuthCheck.ts [hunk 2]
```diff
@@ -29,7 +29,7 @@ const useAuthCheck = (path: Path) => {
     }
 
     checkAuth();
-  }, [navigate]);
+  }, [navigate, path]);
```
Comment (Asmaa-204) [inline app/src/features/authentication/login/hooks/useAuthCheck.ts:32]: this only happens on the initial render

File: app/src/features/authentication/signup/services/apiConfirmCode.ts [hunk 3]
```diff
@@ -6,7 +6,7 @@ export async function sendEmailVerification(email: string) {
     headers: {
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(email),
+    body: JSON.stringify({ email: email }),
```
Comment (Asmaa-204) [inline app/src/features/authentication/signup/services/apiConfirmCode.ts:9]: you can use JSON.stringify({email}) as they have the same name

File: app/src/pages/Signup.tsx [hunk 4]
```diff
@@ -36,8 +36,6 @@ const AnimatedTitle = styled(Heading).attrs({ as: "h1" })`
 `;
 
 function Signup() {
-  useAuthCheck("/signup");
```
Comment (Asmaa-204) [inline app/src/pages/Signup.tsx:39]: ya laaaahwiiii

File: app/src/features/authentication/login/hooks/useAuthCheck.ts (hunk 2 above)
Comment (sarah-kamall) [reply app/src/features/authentication/login/hooks/useAuthCheck.ts:32]: i thought we would just add it as jonas says XD

File: app/src/features/authentication/signup/services/apiConfirmCode.ts (hunk 3 above)
Comment (sarah-kamall) [reply app/src/features/authentication/signup/services/apiConfirmCode.ts:9]: it wasn't working for some reason

File: app/src/pages/Signup.tsx (hunk 4 above)
Comment (sarah-kamall) [reply app/src/pages/Signup.tsx:39]: dude think about it , will the user be authenticated if they are trying to signup

Comment (Asmaa-204) [reply app/src/pages/Signup.tsx:39]: it's just for PROTECTING routes rg3eha tani lazem😂

**Review by Asmaa-204 — APPROVED** [review]
🔥🔥🔥🔥🔥

**Pull request merged**